packages/hadron-plugin-manager/lib/plugin-manager.js

```diff
@@ -63,32 +65,11 @@ class PluginManager {
     return [];
   }
 
-  async _loadPluginsFromList(pluginList, appRegistry, apiVersion) {
-    debug('Loading plugins from list', pluginList);
-    try {
-      return await Promise.all(
-        pluginList.map((pluginName) => {
-          return this._loadPlugin(
-            path.dirname(req.resolve(path.join(pluginName, 'package.json'))),
-            appRegistry,
-            apiVersion
-          );
-        })
-      );
-    } catch (e) {
-      debug('Failed to load plugins from list:', e);
-    }
-    return [];
-  }
-
-  async _loadPlugin(pluginPath, appRegistry, apiVersion) {
+  async _loadPlugin(pluginPath, apiVersion) {
     debug('Loading plugin from path', pluginPath);
     try {
       const plugin = new Plugin(pluginPath, apiVersion);
-      plugin.activate(appRegistry);
-      const frozen = Object.freeze(plugin);
-      this.plugins.push(frozen);
-      Action.pluginActivated(frozen);
+      this.plugins.push(plugin);
```

Comment:
I'm a bit confused here - is this plugin array an array of plugins or strings?
Unless I'm missing something it looks like we're using it for both when we previously has `pluginList` storing the names of the plugins and this array was just for the plugins. Now maybe we're keeping both which would make the loop above fail sometimes?

Reply:
It's an array of plugins, no strings involved there 😄 So `this.paths` that is used above is a list to pats where plugins potentially can be provided by the Compass users (third-party plugins), and `this.plugins` that initiated in constructor from the passed argument is already a list of pre-loaded plugins. So with this change plugin manager is more or less responsible only for loading third-party plugins dynamically, everything else is passed to it from Compass directly. Does this makes more sense? Do you think I can rename things or add some comments to plugin manager to make it clearer?

Reply:
Ohh I totally misread and missed the `require` in lists in the creation of the `PluginManager` thanks for clarifying xD
I think the naming makes sense.